ntp-proto/src/timetypes.rs

```diff
@@ -1,45 +1,57 @@
-#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord)]
+use std::ops::Sub;
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord, Default)]
 pub struct NtpTimestamp {
     timestamp: u64,
 }
 
 impl NtpTimestamp {
-    pub(crate) fn from_bits(bits: [u8; 8]) -> NtpTimestamp {
+    pub(crate) const fn from_bits(bits: [u8; 8]) -> NtpTimestamp {
         NtpTimestamp {
             timestamp: u64::from_be_bytes(bits),
         }
     }
 
-    pub(crate) fn to_bits(&self) -> [u8;8] {
+    pub(crate) const fn to_bits(self) -> [u8; 8] {
         self.timestamp.to_be_bytes()
     }
 
     #[cfg(test)]
-    pub(crate) fn from_fixed_int(timestamp: u64) -> NtpTimestamp {
+    pub(crate) const fn from_fixed_int(timestamp: u64) -> NtpTimestamp {
         NtpTimestamp { timestamp }
     }
 }
 
-#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord)]
+impl Sub for NtpTimestamp {
+    type Output = NtpDuration;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        NtpDuration {
+            duration: self.timestamp as i64 - rhs.timestamp as i64,
```

Comment:
This seems scary. The `u64` range is smaller than `i64` on the low side and larger on the high side. So this should probably be a `checked_sub()` or `saturating_sub()` method instead, depending on what we actually need? (Maybe both?)